tpie/pipelining/tokens.h

```diff
@@ -193,6 +197,17 @@ class node_map {
 	///////////////////////////////////////////////////////////////////////////
 	void get_successors(id_t from, std::vector<id_t> & successors, memory_size_type k, bool forward_only=false);
 
+	void forward(std::string key, any_noncopyable && value) {
+		m_pipelineForwards[key] = std::move(value);
+	}
+
+	maybeany_t fetch_maybe(std::string key) {
+		auto it = m_pipelineForwards.find(key);
+		if (it == m_pipelineForwards.end()) {
+			return maybeany_t();
+		}
+		return maybeany_t(it->second);
+	}
 
```

Comment:
`fetch_maybe` is called from https://github.com/thomasmoelhave/tpie/blob/6782409ecb15fd386c79a2560d65b377329e9f12/tpie/pipelining/node.cpp#L235

`forward` is called from https://github.com/thomasmoelhave/tpie/blob/6782409ecb15fd386c79a2560d65b377329e9f12/tpie/pipelining/pipeline.cpp#L148